src/textual/document/_languages.py

```diff
@@ -1,13 +1,43 @@
 BUILTIN_LANGUAGES = sorted(
```

Comment:
I've found the [TheRenegadeCoder/sample-programs](https://github.com/TheRenegadeCoder/sample-programs) repo useful for checking how these languages render.

<details><summary>App for Testing</summary>

<p>

```python
"""
TextArea Example
"""

import os
import pathlib
from traceback import format_exc
from typing import Any, ClassVar

import click

from textual import on
from textual.app import ComposeResult, App
from textual.containers import Horizontal
from textual.widgets import TextArea, DirectoryTree, Header

KNOWN_EXTENSIONS = {
    "bash": [".sh", ".bash"],
    "c": [".c"],
    "c_sharp": [".cs"],
    "cpp": [".cpp", ".hpp", ".h"],
    "css": [".css"],
    "dockerfile": ["Dockerfile"],
    "dot": [".dot"],
    "elisp": [".el"],
    "elixir": [".ex", ".exs"],
    "elm": [".elm"],
    "embedded_template": [".et"],
    "erlang": [".erl", ".hrl"],
    "fortran": [".f", ".f90", ".f95", ".f03", ".f08", ".f18", ".for", ".f77"],
    "go": [".go"],
    "gomod": ["go.mod", "go.sum"],
    "hack": [".hh"],
    "haskell": [".hs"],
    "html": [".html", ".htm"],
    "java": [".java"],
    "javascript": [".js", ".jsx"],
    "jsdoc": [".jsdoc"],
    "json": [".json"],
    "kotlin": [".kt"],
    "make": ["Makefile", "makefile", "GNUmakefile"],
    "markdown": [".md", ".markdown", ".mdown", ".mkdn"],
    "objc": [".m", ".mm"],
    "ocaml": [".ml", ".mli"],
    "php": [".php", ".php5", ".php7"],
    "python": [".py"],
    "r": [".r"],
    "regex": [".regex"],
    "ruby": [".rb", ".erb"],
    "rust": [".rs"],
    "scala": [".scala"],
    "sql": [".sql"],
    "sqlite": [".sqlite"],
    "toml": [".toml"],
    "typescript": [".ts", ".tsx"],
    "yaml": [".yaml", ".yml"],
}

EXTENSION_TO_LANGUAGE_MAPPING = {
    extension: language
    for language, extensions in KNOWN_EXTENSIONS.items()
    for extension in extensions
}


class TextAreaApp(App):
    CSS: ClassVar[
        str
    ] = """
    DirectoryTree {
        width: auto;
    }
    """

    def __init__(self, path: os.PathLike, *args: Any, **kwargs: Any) -> None:
        super().__init__(*args, **kwargs)
        self.path = pathlib.Path(path)
        self.directory_tree = DirectoryTree(path=self.path)
        self.text_area = TextArea(
            theme="monokai",
            soft_wrap=False,
            show_line_numbers=True,
        )

    def compose(self) -> ComposeResult:
        yield Header()
        yield Horizontal(self.directory_tree, self.text_area)

    @on(DirectoryTree.FileSelected)
    def handle_file_selected(self, event: DirectoryTree.FileSelected) -> None:
        self.path = pathlib.Path(event.path)
        try:
            new_text = event.path.read_text()
            extension_mapping = EXTENSION_TO_LANGUAGE_MAPPING.get(
                self.path.suffix.lower()
            )
            filename_mapping = EXTENSION_TO_LANGUAGE_MAPPING.get(self.path.name)
            language = extension_mapping or filename_mapping or "regex"
            self.text_area.text = new_text
            self.text_area.language = language
            self.sub_title = language
        except:  # noqa
            self.text_area.language = "regex"
            self.text_area.text = f"Error: {format_exc()}"
            self.sub_title = "Error"


@click.command()
@click.argument(
    "path", type=click.Path(exists=True, dir_okay=True, file_okay=False), default="."
)
def cli(path: os.PathLike) -> None:
    """
    TextArea CLI
    """
    app = TextAreaApp(path=path)
    app.run()


if __name__ == "__main__":
    cli()

```

</p>

</details>